src/fortrancode.l

```diff
@@ -938,25 +987,37 @@ LANGUAGE_BIND_SPEC BIND{BS}"("{BS}C{BS}(,{BS}NAME{BS}"="{BS}"\""(.*)"\""{BS})?")
                                           yy_pop_state();
                                           YY_FTN_RESET
                                         }
-<Start>"end"{BS}("block"{BS}"data"|{SUBPROG}|"module"|"program"|"enum"|"type"|"interface")?{BS}     {  // Fortran subroutine or function ends
-                                          //cout << "===> end function " << yytext << endl;
+<Start>^{BS}"end"{BS}/(\n|!|;)          { // bare end keyword.
+                                          //cout << "===> bare end keyword " << yytext << endl;
+                                          //printf("===> line number %d bare end keyword |%s|\n",g_yyLineNr,yytext);
```

Comment:
Why is here a begin of line character (`^`)?

Reply:
The ^ character should be removed.  "end" is likely at the beginning of a logical line, but could be preceeded by a line label.